yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala

```diff
@@ -231,7 +231,7 @@ class ExecutorRunnable(
         "2>", ApplicationConstants.LOG_DIR_EXPANSION_VAR + "/stderr")
 
     // TODO: it would be nicer to just make sure there are no null commands here
-    commands.map(s => if (s == null) "null" else s).toList
+    commands.map(s => if (s == null) "null" else s).map(YarnSparkHadoopUtil.escapeForShell).toList
```

Comment:
I wanted to treat all arguments in one batch, that's why I removed all "escapeForShell" invokations before this spot.